HotelReservation/src/main/java/Day.java

```diff
@@ -0,0 +1,11 @@
+enum Day{
+    WEEKEND(1), WEEKDAY(2);
+    private int id;
+
+    Day(int value){
```

Comment:
Why have you added id to Day? Enums are just  constants. Further there does not seem any requirement of adding a private variable id.

Reply:
We need a integer value for each enum, it is used for overriding hashcode function in Request class.

Reply:
Okay. But in that case it is better to write a wrapper class which contains an id and a type as Day
and make that class as the key of hashmap.